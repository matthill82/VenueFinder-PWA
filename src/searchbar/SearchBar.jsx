import React from "react";
import PropTypes from "prop-types";
import _ from "lodash";
import TextField from "material-ui/TextField";
import Button from "material-ui/Button";

const SearchBar = ({ updateSearchbarQuery, getVenues }) => {
  const debounceUpdateSearchbarQuery = _.debounce(updateSearchbarQuery, 250);

  return (
    <div className="search-bar">
      <TextField
        id="search"
        label="Search field"
        data-selector="search-bar__input"
        type="search"
        className="search-bar__input"
        placeholder="Search spots..."
        margin="normal"
        onChange={event => debounceUpdateSearchbarQuery(event.target.value)}
      />
      <Button
        data-selector="search-bar__button"
        className="search-bar__button"
        variant="raised"
        onClick={getVenues}
      >
        Search near me
      </Button>
    </div>
  );
};

SearchBar.propTypes = {
  getVenues: PropTypes.func.isRequired,
  updateSearchbarQuery: PropTypes.func.isRequired
};

export default SearchBar;
