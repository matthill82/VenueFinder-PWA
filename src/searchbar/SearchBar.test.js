/* eslint-disable */

import React from "react";
import SearchBar from "./SearchBar";
import renderer from "react-test-renderer";
import { shallow, mount } from "enzyme";
import TextField from "material-ui/TextField";
import Button from "material-ui/Button";

describe("<SearcBar getValues updateSearchbarQuery />", () => {
  let component;
  const mockClick = jest.fn();
  const mockChange = jest.fn();

  beforeEach(() => {
    component = shallow(
      <SearchBar getVenues={mockClick} updateSearchbarQuery={mockChange} />
    );
  });

  it("renders correctly", () => {
    const tree = renderer
      .create(
        <SearchBar getVenues={mockClick} updateSearchbarQuery={mockChange} />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe("<TextField />", () => {
    it('exists', () => {
      expect(component.find(TextField).length).toEqual(1)
    })

    xit("Should call a updateSearchbarQuery on Change", () => {
      component = mount(
        <SearchBar getVenues={mockClick} updateSearchbarQuery={mockChange} />
      )
      component.find("[data-selector='search-bar__input'] input").simulate("change", { target: { value: 'xyz' } });
      expect(mockChange).toHaveBeenCalled()
    })

  })

  it("renders without crashing", () => {
    expect(component.exists()).toEqual(true);
  });

  it("should have one input", () => {
    expect(component.find(".search-bar__input").length).toEqual(1);
  });

  describe("<Button />", () => {
    it("Should exist", () => {
      expect(component.find(".search-bar__button").length).toEqual(1);
    });

    it("Should call a click handler when clicked", () => {
      component = mount(
        <SearchBar updateSearchbarQuery={mockChange} getVenues={mockClick} />
      );
      component.find("[data-selector='search-bar__button']").first().simulate("click");
      expect(mockClick).toHaveBeenCalled()
    });
  });
});
