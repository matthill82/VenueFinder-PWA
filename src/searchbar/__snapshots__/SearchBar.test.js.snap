// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearcBar getValues updateSearchbarQuery /> renders correctly 1`] = `
<div
  className="search-bar"
>
  <div
    aria-describedby={undefined}
    className="MuiFormControl-root-1 MuiFormControl-marginNormal-2 search-bar__input"
    data-selector="search-bar__input"
    onBlur={[Function]}
    onFocus={[Function]}
  >
    <label
      className="MuiFormLabel-root-11 MuiInputLabel-root-5 MuiInputLabel-formControl-6 MuiInputLabel-animated-9"
      data-shrink={false}
      htmlFor="search"
    >
      Search field
    </label>
    <div
      className="MuiInput-root-17 MuiInput-formControl-18 MuiInput-underline-21"
    >
      <input
        aria-invalid={false}
        aria-required={false}
        autoComplete={undefined}
        autoFocus={undefined}
        className="MuiInput-input-25 MuiInput-inputType-29 MuiInput-inputTypeSearch-30"
        defaultValue={undefined}
        disabled={false}
        id="search"
        name={undefined}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={undefined}
        onKeyUp={undefined}
        placeholder="Search spots..."
        readOnly={undefined}
        required={undefined}
        rows={undefined}
        type="search"
        value={undefined}
      />
    </div>
  </div>
  <button
    className="MuiButtonBase-root-46 MuiButton-root-31 MuiButton-raised-36 search-bar__button"
    data-selector="search-bar__button"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="button"
  >
    <span
      className="MuiButton-label-32"
    >
      Search near me
    </span>
    <span
      className="MuiTouchRipple-root-49"
    />
  </button>
</div>
`;
